lib/sendingDomains.js

```diff
@@ -0,0 +1,176 @@
+    dkim: {}
+  };
+
+  model.domain = input.domainName;
+  model.dkim['private'] = input.privateKey;
+  model.dkim['public'] = input.publicKey;
+  model.dkim.selector = input.selector;
+  model.dkim.headers = input.headers;
+
+  return model;
+};
+
+/**
+ * Private Method for putting URL construction logic into one place
+ *
+ * @returns string Fully constructed URL for contacting the Sending Domains API
+ */
+var constructURL = function () {
+  return config.options.protocol + '://' + config.options.host + (config.options.port ? ':' + config.options.port: '') + '/api/' + config.options.version + '/sending-domains';
+};
+
+/**
+ *  Private Method for issuing GET request to Sending Domains API
+ *
+ *  This method is responsible for getting the collection _and_
+ *  a specific entity from the Sending Domains API
+ *
+ *  If DomainName parameter is omitted, then we fetch the collection
+ *
+ *  @param DomainName string Domain Name of specific Sending Domain to retrieve
+ *  @param callback function
+ */
+var fetch = function(domainName, callback) {
+  var cb = callback
+    , options = {
+      url: constructURL(),
+      headers: {
+        authorization: config.options.key
+      },
+      strictSSL: config.options.strictSSL
+    };
+
+  /*
+   * Check that someone passed in a real Domain Name, if not,
+   * then the domainName parameter is acutally our callback
+   */
+  if (typeof domainName !== 'function' && typeof domainName !== undefined) {
+    options.url = options.url + '/' + domainName;
+  } else {
+    cb = domainName;
+  }
+
+  // Fetch, Fido, Fetch! (We may want to abstract the actual HTTP requests in a helper when we have more libs)
+  request.get(options, function(err, res, body) {
+    if (err) {
+      cb(new Error('Unable to contact Sending Domains API: ' + err));
+    } else if (res.statusCode === 404) {
+      cb(new Error('The specified Domain Name does not exist'));
+    } else if (res.statusCode !== 200) {
+      cb(new Error('Received bad response from Sending Domains API: ' + res.statusCode));
+    } else {
+      cb(null, body);
+    }
+  });
+};
+
+/**
+ *  Private Method for issuing POST/PUT request to Sending Domains API
+ *
+ *  This method is responsible for creating and updating a specific entity
+ *  from the Sending Domains API
+ *
+ *  @param options object Object containing domain body to create/update and update flag
+ *  @param callback function
+ */
+var write = function(options, callback) {
+  var reqOpts = {
+    url: constructURL() + (options.update ? '/' + options.domainBody.domainName : ''),
+    json: toApiFormat(options.domainBody),
+    headers: {
+      authorization: config.options.key
+    },
+    strictSSL: config.options.strictSSL
+  };
+
+  var reqCallback = function(err, res, body) {
+    if (err) {
+      callback(err);
+    } else if (res.statusCode !== 200) {
+      callback(res.body.errors);
+    } else {
+      callback(null, body);
+    }
+  };
+
+  if (options.update) {
```

Comment:
this could be even simpler by passing the request method into the function, but will work for now.  we can clean later